compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3053,25 +3054,57 @@ TR::Register *OMR::X86::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::C
          }
       else if (comp->getSymRefTab()->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicFetchAndAddSymbol))
          {
-         op = node->getChild(1)->getDataType().isInt32() ? TR::InstOpCode::LXADD4MemReg : TR::InstOpCode::LXADD8MemReg;
+         auto dataType = node->getChild(1)->getDataType();
+         if (dataType.isInt8())
+            op = TR::InstOpCode::LXADD1MemReg;
```

Comment:
As these instructions (LXADD1MemReg/LXADD2MemReg, XCHG1MemReg/XCHG2MemReg) have not been used in the compiler before, would you mind adding encoding tests for them?

Reply:
sure thing!